bluepyopt/ephys/create_acc.py

```diff
@@ -0,0 +1,832 @@
+        NMODL mechanism catalogues compiled with modcc
+        ignored_globals (iterable str): Skipped NrnGlobalParameter in decor
+        replace_axon (): Axon replacement morphology
+        create_mod_morph (): Create ACC morphology with axon replacement
+        template_filename (str): file path of the cell.json , decor.acc and
+        label_dict.acc jinja2 templates (with wildcards expanded by glob)
+        template_dir (str): dir name of the jinja2 templates
+        custom_jinja_params (dict): dict of additional jinja2 params in case
+        of a custom template
+    '''
+
+    if pathlib.Path(morphology).suffix.lower() not in ['.swc', '.asc']:
+        raise CreateAccException("Morphology file %s not supported in Arbor "
+                                 " (only supported types are .swc and .asc)."
+                                 % morphology)
+
+    if replace_axon is not None:
+        if not hasattr(arbor.segment_tree, 'tag_roots'):
+            raise NotImplementedError("Need a newer version of Arbor"
+                                      " for axon replacement.")
+        logger.debug("Obtain axon replacement by applying "
+                     "ArbFileMorphology.replace_axon after loading "
+                     "morphology in Arbor.")
+        replace_axon_path = \
+            pathlib.Path(morphology).stem + '_axon_replacement.acc'
+        replace_axon_acc = io.StringIO()
+        arbor.write_component(replace_axon, replace_axon_acc)
+        replace_axon_acc.seek(0)
+
+        if create_mod_morph:
+            modified_morphology_path = \
+                pathlib.Path(morphology).stem + '_modified.acc'
+            modified_morpho = ArbFileMorphology.load(
+                pathlib.Path(morphology_dir).joinpath(morphology),
+                replace_axon_acc)
+            replace_axon_acc.seek(0)
+            modified_morphology_acc = io.StringIO()
+            arbor.write_component(
+                modified_morpho, modified_morphology_acc)
+            modified_morphology_acc.seek(0)
+            modified_morphology_acc = modified_morphology_acc.read()
+        else:
+            modified_morphology_path = None
+            modified_morphology_acc = None
+
+        replace_axon_acc = replace_axon_acc.read()
+    else:
+        replace_axon_path = None
+        modified_morphology_path = None
+
+    templates = _read_templates(template_dir, template_filename)
+
+    default_location_order = list(ArbFileMorphology.region_labels.values())
+
+    template_params = _get_template_params(mechs,
+                                           parameters,
+                                           ignored_globals,
+                                           disable_banner,
+                                           default_location_order,
+                                           _arb_loc_desc)
+
+    if custom_jinja_params is None:
+        custom_jinja_params = {}
```

Comment:
I can't follow for the first line - I think this refers to a bug elsewhere that has already been fixed. The quoted code is here because it's in the same place as for `create_hoc` and the consistency was maintained to allow refactoring the ACC and HOC exporters into a common base exporter framework in the future (significant parts of the implementation are shared). Btw, a shared API as you mentioned in a [previous comment](https://github.com/BlueBrain/BluePyOpt/pull/393#issuecomment-1307315624) is not needed for that.

Reply:
It's good it's good. The code is good.

I am just asking if you could move these two lines to the beginning of the function because this dict is the default value and I prefer it close to the function signature so that someone reading the signature can tell the default value is in fact a dict (not a None).

Reply:
Done in 04ce61439f8942250b71f7b9f684d6040be14773.